portfolio/src/main/webapp/js/custom-script.js

```diff
@@ -0,0 +1,32 @@
+/* exported insertSearch */
+function insertSearch() {
```

Comment:
is there a reason that you're writing to the client using js instead of just coding in the form and div elements directly into the pages? 

for example will there ever be a case where the search icon won't need to be shown

Reply:
+1

Reply:
I want to have control over the contents of searchBar and possibly incorporate predictive results so having it in my js is useful. Additionally, the search bar will be a feature on every page, so this will allow for the method to developed and changed without changing the html on all the pages. The feature is incomplete.

Reply:
1. To your first point - "have control over the contents of searchBar", the skeleton of search bar is static, and contents generating part can be dynamically controller by JS. But here you are using JS to create the skeleton oof the search bar too.

2. I agree with your second point, which is trying to avoid duplicating html elements.

Reply:
Okay thank you Tong, that makes sense.